logs_processor/logs_processor.go

```
@@ -22,103 +21,114 @@ const (
-			err := s3Object.Body.Close()
-			if err != nil {
-				// Handle the error or log it if necessary.
-				logger.Error("Failed to close s3Object.Body", zap.Error(err))
-			}
-		}()
-
-		// Prepare a buffer for the compacted JSON.
-		var compactedBuffer bytes.Buffer
-
-		// Use json.Compact to write the compacted JSON to the buffer.
-		if err := json.Compact(&compactedBuffer, bodyBytes); err != nil {
-			logger.Error("Failed to compact JSON", zap.Error(err))
-			return nil
+		logsJsons = []map[string]interface{}{{fieldMessage: s3Log}}
+		logs = addFieldsAndAppendLogsToFinalList(logs, logsJsons, logger, key, bucket, awsRegion, controlTowerParsing)
+	}
+	return logs
+}
+
+func processJsonLogs(s3Logs []string, logger *zap.Logger, key, bucket, awsRegion string) [][]byte {
+	// Initializing variables
+	logs := make([][]byte, 0)
+
+	// Process logs
+	controlTowerParsing := getControlTowerParsing()
+	for _, s3Log := range s3Logs {
+		if len(s3Log) == 0 {
+			continue
 		}
 
-		// Unmarshal the compacted JSON to a map for manipulation.
+		var logsJsons []map[string]interface{}
 		var logJson map[string]interface{}
-		if err := json.Unmarshal(compactedBuffer.Bytes(), &logJson); err != nil {
-			logger.Error("Failed to unmarshal compacted JSON", zap.Error(err))
-			return nil
+		err := json.Unmarshal([]byte(s3Log), &logJson)
+
+		if err != nil {
+			logger.Error(fmt.Sprintf("Error occurred while trying to marshal json log: %s", err.Error()))
+			logger.Error("Will try to send as a regular string...")
+			logsJsons = []map[string]interface{}{{fieldMessage: s3Log}}
+		} else {
+			logsJsons = append(logsJsons, logJson)
 		}
+		logs = addFieldsAndAppendLogsToFinalList(logs, logsJsons, logger, key, bucket, awsRegion, controlTowerParsing)
+	}
+	return logs
+}
 
-		// Apply the addLogzioFields function.
+func addFieldsAndAppendLogsToFinalList(logs [][]byte, logsJsons []map[string]interface{}, logger *zap.Logger, key, bucket, awsRegion string, controlTowerParsing string) [][]byte {
```

This function should also return an error, or implement better error handling in the loop.